src/tp_list_maker_server.cpp

```diff
@@ -0,0 +1,234 @@
+//I'm here to  S E R V E  you ;) 
+#include "ros/ros.h"
+#include "hide_and_seek/tp_msg.h"
+#include "hide_and_seek/tp_map_srv.h"
+#include "nav_msgs/OccupancyGrid.h"//PICK AN ACTUAL MAP MESSAGE YO MAYBE OCCUPANCY GRID??
+#include "nav_msgs/GetMap.h"
+
+#define NUM_OF_TURNS 8
+
+class Mratplg//Map Reciever and Tourist Point List Generator
+{
+
+public:
+  Mratplg()
+  {
+    tpl_service = n.advertiseService("generate_new_tp_list", generate_new_tpl);
+    map_reciever = n.serviceClient<map_server>("map_server");//this cant be right
+    ROS_INFO("Ready to generate some A1 TPL's");
+    sub = n.subscribe("amcl_pose", 10, &Mratplg::current_pos_retriever, this);
+  }
+      //prototypes bb
+  bool Mratplg::generate_new_tpl(hide_and_seek::tp_map_srv::Request &req, hide_and_seek::tp_map_srv::Response &res);
+  void current_pos_retriever(const geometry_msgs::PoseWithCovarianceStamped::ConstPtr& current_pos);
+private:
+  ros::NodeHandle n;
+  ros::ServiceServer tpl_service;
+  ros::ServiceClient map_reciever;
+  geometry_msgs::PoseWithCovarianceStampedMessage current_position;
+  //ros::ServiceClient current_position;
+  ros::Subscriber sub;
+};
+
+  
+void Mratplg::current_pos_retriever(const geometry_msgs::PoseWithCovarianceStamped::ConstPtr& current_pos){
+  this.current_position = current_pos;
+}//JUST FINISHED THIS 12:36 AM December 2nd
+  
+bool Mratplg::generate_new_tpl(hide_and_seek::tp_map_srv::Request &req, hide_and_seek::tp_map_srv::Response &res)
+{
+  nav_msgs::GetMap get_map_srv;//no values needed since this service only gets the map B)
+  if(map_reciever.call(get_map_srv))
+  {
+    OccupancyGrid map = get_map_srv.response.map;//not positive if this is completely correct
+  }
+
+  //Probs gotta partition the map into easier to manage chunks or this is gonna take a while
+  //perhaps ONLY analyze a certain distance from the current position?
+  int[map.info.width * map.info.height][4] forces_at_play = { };//Seed the documentation for a fly af explanation
+  int distance_to_left_edge = 0;
+  int distance_to_right_edge = map.info.width;
+  for( int i = 0; i < map.info.width * map.info.height; i++)
+  {
+    if(map.data[i] != 100)
```

Comment:
In my code lately, I've been doing 

```c++
if (100 != map.data[I]) {
  /* do stuff */
}
```

since you get a compiler error if you were doing `==` but accidently typed `=`.

Reply:
Also, this single loop nonsense is a little obfuscated...

```c++
for (size_t x = 0; x < map.info.width; ++x) {
  for (size_t y = 0; y < map.info.height; ++y) {
    /* I _think_ this is right -- I'll check the GMapping source. */
    size_t idx = map.info.width * x + y;
  }
}
```

Reply:
Thank you!

Reply:
Haha that looks passive agressive i promis it was genuine